File: .agy/skills/jbr-pricing-auditor/scripts/audit_pricing.py
import asyncio
import json
import argparse
from crawl4ai import AsyncWebCrawler

async def audit_pricing(urls):
    results = []
    print("Initializing Crawl4AI Auditor...")
    async with AsyncWebCrawler(verbose=True) as crawler:
        for url in urls:
            print(f"Auditing pricing for {url}...")
            try:
                result = await crawler.arun(url=url)
                # Keep first 3000 chars of markdown to capture pricing info
                results.append({
                    "url": url,
                    "pricing_context": result.markdown[:3000] if result.markdown else "No content",
                    "success": result.success
                })
            except Exception as e:
                print(f"Error auditing {url}: {e}")
    
    with open("auditor_output.json", "w", encoding="utf-8") as f:
        json.dump(results, f, indent=2)
    print("Auditing complete. Results saved to auditor_output.json")

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Crawl4AI Pricing Auditor")
    parser.add_argument("--urls", type=str, required=True, help="Comma-separated list of URLs to audit")
    args = parser.parse_args()
    
    url_list = [u.strip() for u in args.urls.split(",") if u.strip()]
    asyncio.run(audit_pricing(url_list))

File: .agy/skills/jbr-pricing-auditor/scripts/audit_pricing.js
const puppeteer = require('puppeteer');
const fs = require('fs');

async function auditPricing() {
    const args = process.argv.slice(2);
    const urlArgIndex = args.indexOf('--urls');
    if (urlArgIndex === -1 || !args[urlArgIndex + 1]) {
        console.error("Please provide urls using --urls <comma-separated-urls>");
        process.exit(1);
    }
    const urlList = args[urlArgIndex + 1].split(',').map(u => u.trim()).filter(u => u);
    
    console.log("Initializing Puppeteer Auditor...");
    const browser = await puppeteer.launch({ headless: 'new' });
    const page = await browser.newPage();
    const results = [];

    for (const url of urlList) {
        console.log(`Auditing pricing for ${url}...`);
        try {
            await page.goto(url, { waitUntil: 'networkidle2', timeout: 30000 });
            // Extract the main body text to simulate raw extraction
            const textContent = await page.evaluate(() => document.body.innerText.substring(0, 4000));
            results.push({
                url: url,
                pricing_context: textContent,
                success: true
            });
        } catch (e) {
            console.error(`Error auditing ${url}: ${e.message}`);
            results.push({ url: url, success: false, error: e.message });
        }
    }
    
    await browser.close();
    fs.writeFileSync("auditor_output.json", JSON.stringify(results, null, 2));
    console.log("Auditing complete. Results saved to auditor_output.json");
}

auditPricing();
